web/src/app/rotations/util.js

```diff
@@ -112,3 +112,39 @@ export function reorderList(_items, oldIndex, newIndex) {
   items.splice(newIndex, 0, _items[oldIndex]) // add dest to newIndex position
   return items
 }
+
+export function getNextHandoffs(
+  count,
+  handoffTime,
+  rotationType,
+  shiftLength,
+  inputTimeZone,
+) {
+  const result = []
+  const rotationTypeToLuxonUnit = {
+    hourly: 'hours',
+    daily: 'days',
+    weekly: 'weeks',
+  }
+  const luxonUnit = rotationTypeToLuxonUnit[rotationType]
+  const now = DateTime.utc()
+
+  let nextHandoff = DateTime.fromISO(handoffTime).setZone(inputTimeZone)
+  if (nextHandoff < now) {
+    while (nextHandoff < now) {
```

Comment:
Have you tested the performance impact of these? I'd be curious to see how it behaves if the start time is well in the past, as we allow any valid ISO timestamp be added as the initial handoff time within the DB.

Reply:
Given that the calculation is so complex, maybe it would be worthwhile to have a GraphQL query. As long as it's calculated on a separate bit of code there's going to be a real chance of discrepancies.

Something like:
```gql
nextRotationHandoffTimes(input:{start: "2020-12-01T14:07:48.172Z", zone: "America/Chicago", hours: 24, count: 3})
```

This will be more important in the future when rotations are part of schedule assignments/rules and things like weekday filters need to be taken into account. Best to write & test that code once I think.

Reply:
I don't think it would be an issue and I think we could get away without debouncing for this.

The actual calculation is negligible (most likely < millisecond since it's just a couple of math operations) so we're only talking network round-trip-time; in our environments, roughly 50ms. It's also not required/blocking for the user to continue and is similar to the phone number validation checkmark on those input fields.

Somewhat separately, I feel like it could always be visible, or visible if the start-time has been changed since I feel like that's almost always going to be helpful when updating that field. In any case, if it is showing we could put a spinner on it, in case the user is on a slow connection.

---

Alternatively, if we *do* opt for implementing it twice, take a look at the `timeutils` branch and `Rotation.EndTime` with the dependent code in `util/timeutil`. We could port that to js/luxon and implement the same set of unit tests, but it doesn't seem worth the hassle/maintenance to keep those in sync IMO